src/common/types.cpp

```diff
@@ -424,7 +424,8 @@ string LogicalType::ToString() const {
 		auto &child_types = StructType::GetChildTypes(*this);
 		string ret = "STRUCT(";
 		for (size_t i = 0; i < child_types.size(); i++) {
-			ret += child_types[i].first + " " + child_types[i].second.ToString();
+			ret += KeywordHelper::WriteOptionallyQuoted(child_types[i].first) + " " +
```

Comment:
This should only add quotes to the struct field name, not the actual type, otherwise you get these issues:

```
STRUCT(a "STRUCT(b ""STRUCT(a """"INTEGER"""", b """"VARCHAR"""")"")")	
```